### PR TITLE
Upgrade Kotlin to 1.3 RC

### PR DESCRIPTION
**Pull request by tiembo**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -30,7 +30,7 @@ buildscript {
         gradleVersion = '3.2.0-rc03'
         gsonVersion = '2.8.2'
         junitVersion = '4.12'
-        kotlinVersion = '1.3-M2'
+        kotlinVersion = '1.3.0-rc-57'
```
Comment (XinyueZ) [inline build.gradle]: It works in "eap" repo.

However, it is really strang that this version is in "eap" not in "dev". 
Take a look at [here, 1.3.0-rc-46](https://mvnrepository.com/artifact/org.jetbrains.kotlin/kotlin-gradle-plugin/1.3.0-rc-46)

**Comment by XinyueZ** [comment]
After 1.3.0-rc-57 the whole AS(3.3 alpha11) doesn't work. I can only build in command line and Kotlin files, code autocomplete, everything down.
However, some other projects don't have this problem. 
Anyway, please check with AS(3.3 alpha11).

**Comment by andhie** [comment]
is there any reason that upgrading kotlin to eap version?
should user always use canary builds in order to build this app successfully?

**Comment by XinyueZ** [comment]
I don't think this app must be built by canary. 😋

**Comment by XinyueZ** [comment]
Build with 1.3-M2 is no problem right now.

**Comment by XinyueZ** [comment]
Update to 1.3-rc-57, I have some problems with AS 3.3.0 canary 11 and gradle-3.3.0-alpha11.
After I've done everything, the IDE cannot work, no code-auto-complete, no SDK found etc.

I prefer to wait for Google internal people to find some solutions.

**Comment by andhie** [comment]
i have no problem using 1.3.0-rc-57 with 1.3.0-rc57-Studio3.2.1 plugin (AS3.2.1 RC3) but have red lines  as mentioned when using 1.3M2 with 1.3.0-rc57-Studio3.2.1 plugin

**Comment by tiembo** [comment]
> is there any reason that upgrading kotlin to eap version?
> should user always use canary builds in order to build this app successfully?

@andhie originally I thought it'd be a good idea to also have the preview version of Kotlin in this app, but after some thought I think it's better to use as many stable libraries as possible.

Because this is primarily a Jetpack sample most of those libraries will probably be preview versions to keep in line with the latest updates, but items like Kotlin and Android Studio version should be stable, and only use preview versions when the Jetpack libraries requires them to.

Since Kotlin 1.3 is coming soon it probably doesn't make sense to downgrade the Kotlin version to 1.2.x and then re-upgrade to 1.3 later.  I'll go ahead with this PR and then not use preview versions of Kotlin (after 1.3) unless it's needed.

Re: Android 3.3 and Kotlin issues - thanks for filing the bug @XinyueZ.  We'll keep Sunflower with AS 3.2 for now.

**Comment by XinyueZ** [comment]
@tiembo Actually I am crazy coding with different versions of IDE 😆

**Comment by VitaFitSoft** [comment]
Intellisense not working and no description when hovering over variable or functions in Studio 3.3. Also getting constant ice errors.

com.google.common.util.concurrent.UncheckedExecutionException: java.io.IOException: File format error reading /Users/user/.gradle/caches/transforms-1/files-1.1/constraintlayout-2.0.0-alpha2.aar/d84d7c6d9269ed46ddb3020be98c7f6d/R.txt line 1743: 'int[] styleable GradientColor { 0x0101020b, 0x010101a2, 0x010101a3, 0x0101019e, 0x01010512, 0x01010513, 0x010101a4, 0x0101019d, 0x01010510, 0x01010511, 0x01010201, 0x010101a1 }'
	at com.android.utils.concurrency.CacheUtils.getAndUnwrap(CacheUtils.kt:37)
	at com.android.tools.idea.res.ProjectLightResourceClassService.getAarRClasses(ProjectLightResourceClassService.kt:172)
	at com.android.tools.idea.res.ProjectLightResourceClassService.getAarRClasses$default(ProjectLightResourceClassService.kt:169)
	at com.android.tools.idea.res.ProjectLightResourceClassService.getLightRClassesAccessibleFromModule(ProjectLightResourceClassService.kt:122)
	at org.jetbrains.android.AndroidResolveScopeEnlarger.getAdditionalResolveScope(AndroidResolveScopeEnlarger.kt:54)
	at org.jetbrains.kotlin.idea.caches.project.IdeaModuleInfosKt.enlargedSearchScope(IdeaModuleInfos.kt:65)
	at org.jetbrains.kotlin.idea.caches.project.IdeaModuleInfosKt.access$enlargedSearchScope(IdeaModuleInfos.kt:1)
	at org.jetbrains.kotlin.idea.caches.project.ModuleProductionSourceInfo.contentScope(IdeaModuleInfos.kt:173)
	at org.jetbrains.kotlin.idea.caches.resolve.KotlinCacheServiceImpl.filterNotInProjectSource(KotlinCacheServiceImpl.kt:463)
	at org.jetbrains.kotlin.idea.caches.resolve.KotlinCacheServiceImpl.getFacadeToAnalyzeFiles(KotlinCacheServiceImpl.kt:418)
	at org.jetbrains.kotlin.idea.caches.resolve.KotlinCacheServiceImpl.getResolutionFacade(KotlinCacheServiceImpl.kt:78)
	at org.jetbrains.kotlin.idea.caches.resolve.ResolutionUtils.getResolutionFacade(resolutionApi.kt:37)
	at org.jetbrains.kotlin.idea.caches.resolve.ResolutionUtils.analyze(resolutionApi.kt:112)
	at org.jetbrains.kotlin.idea.references.AbstractKtReference$Companion$KotlinReferenceResolver.resolveToPsiElements(KtReference.kt:81)
	at org.jetbrains.kotlin.idea.references.AbstractKtReference$Companion$KotlinReferenceResolver.resolve(KtReference.kt:119)
	at org.jetbrains.kotlin.idea.references.AbstractKtReference$Companion$KotlinReferenceResolver.resolve(KtReference.kt:77)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:150)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:180)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:172)
	at org.jetbrains.kotlin.idea.references.AbstractKtReference.multiResolve(KtReference.kt:49)
	at com.intellij.psi.impl.source.resolve.reference.impl.PsiMultiReference.resolves(PsiMultiReference.java:52)
	at com.intellij.psi.impl.source.resolve.reference.impl.PsiMultiReference.lambda$static$0(PsiMultiReference.java:36)
	at java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
	at java.util.TimSort.sort(TimSort.java:220)
	at java.util.Arrays.sort(Arrays.java:1438)
	at com.intellij.psi.impl.source.resolve.reference.impl.PsiMultiReference.chooseReference(PsiMultiReference.java:72)
	at com.intellij.psi.impl.source.resolve.reference.impl.PsiMultiReference.resolve(PsiMultiReference.java:122)
	at com.intellij.codeInsight.TargetElementUtil.doGetReferenceOrReferencedElement(TargetElementUtil.java:382)
	at com.intellij.codeInsight.TargetElementUtil.getReferenceOrReferencedElement(TargetElementUtil.java:359)
	at com.intellij.codeInsight.TargetElementUtil.doFindTargetElement(TargetElementUtil.java:231)
	at com.intellij.codeInsight.TargetElementUtil.findTargetElement(TargetElementUtil.java:203)
	at com.intellij.codeInsight.documentation.DocumentationManager.findTargetElementUnsafe(DocumentationManager.java:637)
	at com.intellij.codeInsight.documentation.DocumentationManager.findTargetElement(DocumentationManager.java:609)
	at com.intellij.codeInsight.documentation.QuickDocOnMouseOverManager$MyShowQuickDocRequest.lambda$run$0(QuickDocOnMouseOverManager.java:296)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1164)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:70)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:123)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:580)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:525)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:85)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:144)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:112)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:70)
	at com.intellij.codeInsight.documentation.QuickDocUtil.runInReadActionWithWriteActionPriorityWithRetries(QuickDocUtil.java:80)
	at com.intellij.codeInsight.documentation.QuickDocOnMouseOverManager$MyShowQuickDocRequest.run(QuickDocOnMouseOverManager.java:294)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:232)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:405)
	at com.intellij.util.Alarm$Request.access$700(Alarm.java:330)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:371)
	at com.intellij.util.Alarm$Request.run(Alarm.java:382)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:228)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:226)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$100(BoundedTaskExecutor.java:26)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2$1.run(BoundedTaskExecutor.java:199)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:229)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:193)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: File format error reading /Users/user/.gradle/caches/transforms-1/files-1.1/constraintlayout-2.0.0-alpha2.aar/d84d7c6d9269ed46ddb3020be98c7f6d/R.txt line 1743: 'int[] styleable GradientColor { 0x0101020b, 0x010101a2, 0x010101a3, 0x0101019e, 0x01010512, 0x01010513, 0x010101a4, 0x0101019d, 0x01010510, 0x01010511, 0x01010201, 0x010101a1 }'
	at com.android.ide.common.symbols.SymbolIo$SymbolLineReader.readLines(SymbolIo.java:322)
	at com.android.ide.common.symbols.SymbolIo.read(SymbolIo.java:145)
	at com.android.ide.common.symbols.SymbolIo.readFromAapt(SymbolIo.java:110)
	at com.android.tools.idea.res.NonNamespacedAarPackageRClass.<init>(AarRClasses.kt:92)
	at com.android.tools.idea.res.ProjectLightResourceClassService$getAarRClasses$2.invoke(ProjectLightResourceClassService.kt:183)
	at com.android.tools.idea.res.ProjectLightResourceClassService$getAarRClasses$2.invoke(ProjectLightResourceClassService.kt:59)
	at com.android.utils.concurrency.CacheUtils$sam$java_util_concurrent_Callable$0.call(CacheUtils.kt)
	at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4869)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3523)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2249)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2132)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2045)
	at com.google.common.cache.LocalCache.get(LocalCache.java:3962)
	at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4864)
	at com.android.utils.concurrency.CacheUtils.getAndUnwrap(CacheUtils.kt:33)
	... 61 more
Caused by: java.io.IOException: Values string { 0x01010532, 0x0101053f, 0x01010533, 0x7f04009c, 0x7f0400a4, 0x7f0400a5, 0x7f0400a6, 0x7f040189 } should have 10 item(s).
	at com.android.ide.common.symbols.SymbolUtils.failParseArrayLiteral(SymbolUtils.kt:404)
	at com.android.ide.common.symbols.SymbolUtils.parseArrayLiteral(SymbolUtils.kt:392)
	at com.android.ide.common.symbols.SymbolIo$SymbolLineReader.handleStyleable(SymbolIo.java:383)
	at com.android.ide.common.symbols.SymbolIo$SymbolLineReader.readLines(SymbolIo.java:285)
	... 75 more

**Comment by XinyueZ** [comment]
@VitaFitSoft see https://github.com/googlesamples/android-sunflower/pull/192#issuecomment-423931936

**Comment by XinyueZ** [comment]
@VitaFitSoft AS 330 Alpha 13 is good, no more problems.

**Pull request merged**